internal/server/providers.go

```diff
@@ -156,17 +160,45 @@ func (a *API) UpdateProvider(c *gin.Context, r *api.UpdateProviderRequest) (*api
 	}
 	provider.Kind = kind
 
-	if err := a.setProviderInfoFromServer(c, provider); err != nil {
+	if err := overrideManagedProviderFields(c, provider); err != nil {
 		return nil, err
 	}
 
+	if !provider.Managed {
+		if err := a.setProviderInfoFromServer(c, provider); err != nil {
+			return nil, err
+		}
+	}
+
 	if err := access.SaveProvider(c, provider); err != nil {
 		return nil, err
 	}
 
 	return provider.ToAPI(), nil
 }
 
+// overrideManagedProviderFields checks if an existing provider is managed on update and overrides managed fields
+func overrideManagedProviderFields(c *gin.Context, p *models.Provider) error {
+	existing, err := access.GetProvider(c, p.ID)
+	if err != nil {
+		return err
+	}
+	if existing.Managed {
+		// can't update managed client ID, secret, URL, or API details
```

Comment:
Which fields can be updated? Is it only the name?

Would it be easier and safer to explicitly allow updates of specific fields (`Name`), rather than default to allow and having to remember to exclude the fields here every time we add a field?

Reply:
This went away with one Google provider being shared for all orgs